crates/ruff/tests/snapshots/integration_test__rule_f401.snap

```diff
@@ -34,12 +34,29 @@ marking it as unused, as in:
 from module import member as member
 ```
 
+Alternatively, you can use `__all__` to declare a symbol as part of the module's
+interface, as in:
+
+```python
+# __init__.py
+import some_module
+
+__all__ = [ "some_module"]
+```
+
 ## Fix safety
 
-When `ignore_init_module_imports` is disabled, fixes can remove for unused imports in `__init__` files.
-These fixes are considered unsafe because they can change the public interface.
+Fixes to remove unused imports are safe, except in `__init__.py` files.
+
+Applying fixes to `__init__.py` files is currently in preview. The fix offered depends on the
+type of the unused import. Ruff will suggest a safe fix to export first-party imports with
+either a redundant alias or, if already present in the file, an `__all__` entry. If multiple
+`__all__` declarations are present, Ruff will not offer a fix. Ruff will suggest an unsafe fix
+to remove third-party and standard library imports -- the fix is unsafe because the module's
+interface changes.
```

Comment:
I think it's good that ruff has different handling of first vs third party imports, but changing behavior based on the name of the module (containing the import) I find unintuitive and I argue is unpythonic.

By that, I mean: if I have `mod/__init__.py`, is the fix any more or less safe than if I instead named it `mod.py`? No — the `__init__.py` is just an implementation detail and in either case, the module's interface changes with _any_ import changes.

It's not a universally-accepted convention that `__init__.py` should store your entire public API.
For example, it's hinders import times for users of a large package, if the entire public API is placed in `__init__.py`, because then any import of a _submodule_ will trigger an import `__init__.py` and thus an import of _every_ module, if _everything_ is reexported.

At most, we have the typing docs ([Typing documentation: interface conventions](https://typing.readthedocs.io/en/latest/source/libraries.html#library-interface-public-and-private-symbols)) because there's nothing I can find on python.org that discusses this. But given the typing docs guidance, I would find it slightly easier to teach that:

* third party imports: _always_ safe to remove
* first party imports: _unsafe_ to remove if contained within the same package (similar to relative import convention from typing spec). Safe otherwise.

Reply:
Thanks for your feedback @Hnasar — would you mind opening a new issue to discuss this preview behavior? I don't think this pull request is a great place for it since it's just updating the docs.